### PR TITLE
Fix: handle initialisation of Intrinsic elemental functions to compile time constant

### PR DESCRIPTION
**Pull request by HarshitaKalani**
Towards: https://github.com/lfortran/lfortran/issues/3860#issuecomment-2051719754

### CODE REVIEW

**Comment by HarshitaKalani** [comment]
```fortran
program main
implicit none

integer, parameter :: new_len = len(adjustl('a'))

print *, new_len
if (new_len /= 1) error stop

end program

```
```console
$ lfortran a.f90
Internal Compiler Error: Unhandled exception
^C
```
The `len` function has an issue different that this one. Will be handling that in next PR.

File: integration_tests/intrinsics_219.f90 [hunk 1]
```diff
@@ -0,0 +1,4 @@
+program intrinsics_219
+    integer :: x = sum([1, 2, 3])
+    if (x /= 6) error stop
```
Comment (Pranavchiku) [inline integration_tests/intrinsics_219.f90:3]:
```suggestion
    integer, parameter :: x = sum([1, 2, 3])
    if (x /= 6) error stop
```

Does this work?

**Review by Pranavchiku — APPROVED** [review]
Left comment, rest is fine!

File: integration_tests/intrinsics_219.f90 (hunk 1 above)
Comment (HarshitaKalani) [reply integration_tests/intrinsics_219.f90:3]: `integer, parameter :: x = sum([1, 2, 3])`
This gives correct answer but the error stop condition throws error due to type mismatch. Will create an another issue for that.

**Pull request merged**